### DISCUSSION TITLE
How to use yapf as formatter of Python

### DISCUSSION THREAD

**Discussion by MoienBowen** [zed-industries/zed#6467] [Feature Requests]
I followed the documentation, it works with your suggested formater `black` as 
```
  "language_overrides": {
    "Python": {
      "format_on_save": {
        "external": {
          "command": "black",
          "arguments": ["-"]
        }
      }
    }
  }
```

So I tried with `yapf`, it works without any argument as

```
  "language_overrides": {
    "Python": {
      "format_on_save": {
        "external": {
          "command": "yapf",
          "arguments": []
        }
      }
    }
  }
```

However, I would like to change the tab length as 2, hence, I need to add the option `--style='{based_on_style: pep8, indent_width: 2}'`. It does not work when I modify the settings.json as follows

```
  "language_overrides": {
    "Python": {
      "format_on_save": {
        "external": {
          "command": "yapf",
          "arguments": [
            "--style='{based_on_style: pep8, indent_width: 2}'"
          ]
        }
      }
    }
  }
```

Is there any one who could help me on this, please?

hits:

`yapf  --style='{based_on_style: pep8, indent_width: 2}' -I filename`
`echo filename | xargs yapf  --style='{based_on_style: pep8, indent_width: 2}' -i`

**Comment by HBDE-work** [comment]
_Note that i build my Zed from master, so I'm currently running Version 0.187.0_

I followed https://zed.dev/docs/configuring-zed#formatter

But i cannot make it work with your arguments, too.

Instead i have a `.style.yapf` file in my repository root and then using this setting snippet:

```json
   "languages": {
        "Python": {
            "language_servers": ["!pylsp", "pyright"],
            "format_on_save": "on",
            "formatter": {
                "external": {
                    "command": "yapf",
                    "arguments": null
                }
            }
        }
    }
```

which does work!

I don't know if i can access any log from within Zed to see the output of yapf. I guess with arguments the command fails.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
